libkbfs/interfaces.go

```diff
@@ -1447,3 +1447,177 @@ type RekeyQueue interface {
+	// TLF is public, or if the given key generation is invalid. Equivalent to:
+	//
+	//   kids, err := GetDeviceKIDs(keyGen, user)
+	//   return (err == nil) && (len(kids) > 0)
+	HasKeyForUser(keyGen KeyGen, user keybase1.UID) bool
+	// GetTLFCryptKeyParams returns all the necessary info to construct
+	// the TLF crypt key for the given key generation, user, and device
+	// (identified by its crypt public key), or false if not found. This
+	// returns an error if the TLF is public.
+	GetTLFCryptKeyParams(keyGen KeyGen, user keybase1.UID, key CryptPublicKey) (
+		TLFEphemeralPublicKey, EncryptedTLFCryptKeyClientHalf,
+		TLFCryptKeyServerHalfID, bool, error)
+	// IsValidAndSigned verifies the BareRootMetadata given the current
+	// user and device (identified by the KID of the device verifying
+	// key), checks the writer signature, and returns an error if a
+	// problem was found.
+	IsValidAndSigned(codec Codec, crypto cryptoPure,
+		currentUID keybase1.UID, currentVerifyingKey VerifyingKey) error
+	// LastModifyingWriter return the UID of the last user to modify the writer metadata.
+	LastModifyingWriter() keybase1.UID
+	// LastModifyingWriterKID returns the KID of the last device to modify the writer metadata.
+	LastModifyingWriterKID() keybase1.KID
+	// LastModifyingUser return the UID of the last user to modify the any of the metadata.
+	GetLastModifyingUser() keybase1.UID
+	// RefBytes returns the number of newly referenced bytes introduced by this revision of metadata.
+	RefBytes() uint64
+	// UnrefBytes returns the number of newly unreferenced bytes introduced by this revision of metadata.
+	UnrefBytes() uint64
+	// DiskUsage returns the estimated disk usage for the folder as of this revision of metadata.
+	DiskUsage() uint64
+	// RevisionNumber returns the revision number associated with this metadata structure.
+	RevisionNumber() MetadataRevision
+	// BID returns the per-device branch ID associated with this metadata revision.
+	BID() BranchID
+	// GetPrevRoot returns the hash of the previous metadata revision.
+	GetPrevRoot() MdID
+	// IsUnmergedSet returns true if the unmerged bit is set.
+	IsUnmergedSet() bool
+	// GetSerializedPrivateMetadata returns the serialized private metadata as a byte slice.
+	GetSerializedPrivateMetadata() []byte
+	// GetSerializedWriterMetadata serializes the underlying writer metadata and returns the result.
+	GetSerializedWriterMetadata(codec Codec) ([]byte, error)
+	// GetWriterMetadataSigInfo returns the signature info associated with the the writer metadata.
+	GetWriterMetadataSigInfo() SignatureInfo
+	// Version returns the metadata version.
+	Version() MetadataVer
+	// Mutable returns the mutable BareRootMetadata interface associated with this immutable version.
+	// This is a complete hack for now to get around the fact that ImmutableRootMetadata embeds a
+	// RootMetadata pointer and some places in the code create a RootMetadata from an ImmutableBareRootMetadata.
+	Mutable() BareRootMetadata
```

Comment:
I don't think you need this. Type assertions are type-safe enough, e.g.

```
mutableMD, ok := irmd.(MutableBareRootMetadata)
if !ok {
  // return an error
}
```
...